### PR TITLE
feat: Colored Error, Warning and Debug prefix.

### PR DESCRIPTION
**Pull request by SandPod**
issue: https://github.com/serverpod/serverpod/issues/1042

## Adds:

Colors the prefixes for Error, Warning and Debug messages so that they are easier to spot.

### After
image
image

## Fixes:
Word wrap didn't calculate the correct width of the line. This fixes that and ensures that any trimmable characters in the beginning of a line are preserved.

### Before
Screenshot 2023-07-28 at 10 15 58

### After
image

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

None

### CODE REVIEW

File: tools/serverpod_cli/lib/src/logger/loggers/std_out_logger.dart [hunk 1]
```diff
@@ -233,12 +245,34 @@ String _wrapText(String text, int columnWidth) {
...
   List<String> outLines = [];
   for (var line in textLines) {
-    outLines.add(line.wordWrap(width: columnWidth));
+    var replaceFirstChar = _hasLeadingTrimCharacter(line);
+    // wordWrap(...) uses trim as part of its implementation which removes all
+    // leading trimmable characters.
+    // In order to preserve them we temporarily replace the first char with a
+    // non trimmable character.
+    if (replaceFirstChar) {
+      line = '@${line.substring(1)}';
```
Comment (Isakdl) [inline tools/serverpod_cli/lib/src/logger/loggers/std_out_logger.dart:254]: Question: Is the first char guaranteed to be a white space? If not we would lose that information here.

**Review by Isakdl — APPROVED** [review]
LGTM :) One question on the formatting if it holds true I think this is good to go.

File: tools/serverpod_cli/lib/src/logger/loggers/std_out_logger.dart (hunk 1 above)
Comment (SandPod) [reply tools/serverpod_cli/lib/src/logger/loggers/std_out_logger.dart:254]: Yes, the check we do above trims the first character of the text. And if the result of the trim is empty then we know that the first char would disappear once we call the wordWrap method of superstring. Therefore we can safely replace the symbol.

**Review by vlidholt — APPROVED** [review]
LGTM.

A thought though, maybe it would nicer to skip the `WARNING:` / `ERROR:` texts and just color the whole line? It looks a bit messy with multiple errors. Maybe have a look at other tools, e.g. the Flutter command line. What's their warning/error output like?

**Review by vlidholt — APPROVED** [review]
LGTM :)

**Pull request merged**